.github/workflows/pipeline.yml

```diff
@@ -334,7 +338,7 @@ jobs:
   deploy:
     needs: [setup, build]
     name: 'Deploy the application'
-    if: ${{ contains(inputs.ci_steps, 'deploy') }}
+    if: ${{ always() && inputs.build_cache != '' || contains(inputs.ci_steps, 'deploy') }}
```

Comment:
What does `always()` do?

Reply:
In this case, it should always run deploy if build_cache input is defined. It's a must since the build is in a needs block. I still need to test it.